lib/pagetience.rb

```diff
@@ -2,28 +2,52 @@
 require 'pagetience/timer'
 require 'pagetience/version'
 
+require 'pagetience/platforms/base'
+require 'pagetience/platforms/page-object'
+
 module Pagetience
-  SUPPORTED_ELEMENT_LIBS = [PageObject]
+  module ClassMethods
+    def required(*elements)
+      elements.keep_if { |e| e.is_a? Symbol }
+      define_method('_required_elements') do
+        elements
+      end
+    end
+
+    def waiting(timeout, polling=1)
+      define_method('_waiting_timeout') do
+        timeout
+      end
+      define_method('_waiting_polling') do
+        polling
+      end
+    end
+  end
+
+  SUPPORTED_PLATFORMS = {
+      ancestors: [
+          {name: 'page-object', module: PageObject, platform: Pagetience::Platforms::PageObjectGem}
+      ]
+  }
```

Comment:
Maybe this can be cleaned up/simplified like...

```ruby
SUPPORTED_PLATFORMS = [ Pagetience::Platforms::PageObjectGem, ... ]

@element_platform = SUPPORTED_PLATFORMS.find { |p| p.present? }

# ...
def present?
  # determine if this platform is present however it should be
end
```